### ISSUE TITLE
Feature Request (For translators): Option to show the table of associations

### ISSUE THREAD

**Issue by openbibles** [rubiot/ibiblia#7]
Since, you already provided a way to export associations why not give the option to show them while working on the associations which could be useful for translation purposes. I have my own table of English word to Tagalog conversion, but since its built in to this program why not just use it?

**Comment by rubiot** [comment]
Yes, this would be great. I'll add that to my todo list.

**Comment by rubiot** [comment]
An **experimental** implementation of this feature is available with the new release [v1.0.0.48](https://github.com/rubiot/ibiblia/milestone/1). Alt+Click a word to see all the translations used for it (the word by itseflt or within an expression) in the dictionary popup. Please check it and give me some feedback.

**Comment by malayangbiblia** [comment]
Sweet!

**Comment by malayangbiblia** [comment]
Pop up works and it shows a bullet points of all the words associated. But the pop-up doesn't seem to go away.

**Comment by malayangbiblia** [comment]
Can I suggest some data to put in the pop-up? Here is the list:
For the Source Text pop-up:
1. Associated words from the Destination text appears
2. There will be number of instances that word appears next to it just like - Hesus (5)
3. The verses where those words appear will be listed at the bottom of or next to each word like:
Hesus (5) - Joh 1:1, Rev 1:1, Heb 1:1, etc.

For the Destination Text pop-up:
1. Associated words from the Source text appears
2. There will be number of instances that word appears next to it just like - Jesus (5)
3. The verses where those words appear will be listed at the bottom of or next to each word like:
Jesus (5) - Joh 1:1, Rev 1:1, Heb 1:1, etc.

**Comment by rubiot** [comment]
> 1. the pop-up doesn't seem to go away except if I restart it.

I didn't do many tests, but the intend way to use it is: you Alt-Click a word (and don't move your mouse), the popup pops up, you release the Alt key, and once you've read it you just move your mouse away from the popup and it should disappear.

> 2. if I put the dictionary then its somehow overlapped by the pop up dictionary definition because its using both the Alt key.

Yes, I know. I've ran out of modifiying keys... haha.

> Would it be feasible and better if you just put a section on the toolbar with 2 radio buttons inside that's labeled at the top "Pop-up". The one button says "Dictionary" and the other one says "Associations". Then if D is selected, only the Dictionary entry will pop up with the Alt+hover combo, but if A is selected then only the Associations word entry will pop up with the Alt+hover combo. Does that make sense?

This would make sense if the chooses an option and sticks to it. I'd like to have both the dictionary and the associations readily available. Maybe a better idea would be to add a permanent tab to the popup itself. Then if the user didn't configure any dictionary he would see only the associations tab on the popup. If the configured a dictionary, he would see two tabs on the popup. What do you say?

**Comment by rubiot** [comment]
> Can I suggest some data to put in the pop-up? Here is the list:
> For the Source Text pop-up:
> 
> 1. Associated words from the Destination text appears
> 2. There will be number of instances that word appears next to it just like - Hesus (5)
> 3. The verses where those words appear will be listed at the bottom of or next to each word like:
>    Hesus (5) - Joh 1:1, Rev 1:1, Heb 1:1, etc.
> 
> For the Destination Text pop-up:
> 
> 1. Associated words from the Source text appears
> 2. There will be number of instances that word appears next to it just like - Jesus (5)
> 3. The verses where those words appear will be listed at the bottom of or next to each word like:
>    Jesus (5) - Joh 1:1, Rev 1:1, Heb 1:1, etc.

Today iBiblia doesn't have all this information readily available. It has when it generates the concordance, but only after going through all verses summarizing all associations. So, what I'm thinking of doing right now it to add a word count or a percentage next to each bullet point. Another thing it to divide the bullet point in two sections: word-to-word associations and expression associations (when the word wasn't used alone).

Of course all these could be done, but it would require much effort. I'll have to postpone it for now. I hope you understand. :)

**Comment by malayangbiblia** [comment]
> This would make sense if the chooses an option and sticks to it. I'd like to have both the dictionary and the associations readily available. Maybe a better idea would be to add a permanent tab to the popup itself. Then if the user didn't configure any dictionary he would see only the associations tab on the popup. If the configured a dictionary, he would see two tabs on the popup. What do you say?
Yes, its pretty much the same. whichever is easier for to code.

**Comment by malayangbiblia** [comment]
> Today iBiblia doesn't have all this information readily available. It has when it generates the concordance, but only after going through all verses summarizing all associations. So, what I'm thinking of doing right now it to add a word count or a percentage next to each bullet point. Another thing it to divide the bullet point in two sections: word-to-word associations and expression associations (when the word wasn't used alone).

Of course all these could be done, but it would require much effort. I'll have to postpone it for now. 

> 

I agree. Its really not that important. What's important are the bug fixes.

**Comment by openbibles** [comment]
I think this feature serves its purpose and doesn't need any additional formatting or extra structure since all these info can be exported into the analytical dictionary module.

**Comment by openbibles** [comment]
I was thinking if there is a way to clean up the associations because when I try to review my work, there are associations that I am revising (removing and adding). But it seems that the unwanted data or associated words in the table remain. Will the "Recreate Association Memory" feature fix this?

**Comment by rubiot** [comment]
iBiblia should keep the association memory up-to-date even after you undo associations. Maybe you've found a bug. Anyway, "Recreate association memory" will fix that.

**Comment by openbibles** [comment]
Cool! thanks bro! Good to hear from you again!

**Comment by rubiot** [comment]
The newest release now shows **all** alternatives. It was showing only the translations done to the exact inflection of the word. I've added a count for each alternative as well.

**Comment by rubiot** [comment]
I still need to differentiate the cases in which the word was translated alongside other words - i.e. within an expression.

**Comment by openbibles** [comment]
Great!
